RevenueCatUI/Views/TimelineView.swift

```diff
@@ -0,0 +1,322 @@
+//
+//  File.swift
+//  
+//
+//  Created by Guillaume LAURES on 15/05/2024.
+//
+
+import SwiftUI
+import UIKit
+
+extension NSLayoutConstraint.Axis {
+    
+    var opposite: NSLayoutConstraint.Axis {
+        switch self {
+        case .vertical:
+            return .horizontal
+        case .horizontal:
+            return .vertical
+        @unknown default:
+            return self
+        }
+    }
+}
+
+@available(iOS 13.0, *)
+extension HorizontalAlignment {
+    private enum IconAlignment: AlignmentID {
+        static func defaultValue(in dimensions: ViewDimensions) -> CGFloat {
+            return dimensions[HorizontalAlignment.center]
+        }
+    }
+    static let iconAlignment = HorizontalAlignment(IconAlignment.self)
+}
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+@available(macOS, unavailable)
+@available(tvOS, unavailable)
+struct TimelineView: View {
+    
+    enum Colors {
+        static let green = Color(red: 8.0 / 255.0, green: 195.0 / 255.0, blue: 130.0 / 255.0)
+        static let linearityOrange = Color(red: 253.0 / 255.0, green: 122.0 / 255.0, blue: 15.0 / 255.0)
+        static let link = Color(uiColor: UIColor(dynamicProvider: { trait in
+            if trait.userInterfaceStyle == .light {
+                return UIColor(red: 225.0 / 255.0, green: 225.0 / 255.0, blue: 225.0 / 255.0, alpha: 1)
+            } else {
+                return UIColor(red: 174.0 / 255.0, green: 174.0 / 255.0, blue: 178.0 / 255.0, alpha: 1)
+            }
+        }))
+        static let bell = Color(uiColor: UIColor(dynamicProvider: { trait in
+            if trait.userInterfaceStyle == .light {
+                return .black
+            } else {
+                return UIColor(Colors.link)
+            }
+        }))
+        static let iconBackground = Color(uiColor: UIColor(dynamicProvider: { trait in
+            if trait.userInterfaceStyle == .light {
+                return .white
+            } else {
+                return .secondarySystemBackground
+            }
+        }))
+    }
+    
+    let stepConfigurations: [TimelineStepView.Configuration]
+    
+    @Environment(\.horizontalSizeClass)
+    var horizontalSizeClass
+    var axis: NSLayoutConstraint.Axis {
+        horizontalSizeClass == .regular ? .vertical : .horizontal
+    }
+    
+    var body: some View {
+        if axis == .horizontal {
+            HStack(spacing: 0) {
+                content
+            }
+        } else {
+            VStack(alignment: .leading, spacing: 0) {
+                content
+            }
+        }
+    }
+    
+    @ViewBuilder
+    var content: some View {
+        ForEach(stepConfigurations) {
+            TimelineStepView(configuration: $0, axis: axis.opposite)
+        }
+    }
+}
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+@available(macOS, unavailable)
+@available(tvOS, unavailable)
+struct TimelineStepView: View {
```

Comment:
private?

Reply:
If I do that then I have a problem because `TimelineView` exposes `TimelineStepView.Configuration`. Moving `TimelineStepView.Configuration` into `TimelineView` also does not work because then I have a problem with `TimelineStepView.LinkPosition`.

So not sure what's the best here.